### PR TITLE
feat: add other interfaces from the spec for discovery

### PR DESCRIPTION
**Pull request by loopingz**
Signed-off-by: Remi Cattiau <remi@cattiau.com>

## Proposed Changes

- Add the interface from the Discovery spec
- Another package including Discovery service and Subscription service will be created
- Clean formatting of `examples/express-ex/index.js` example

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { CloudEvent, Version } from "./event/cloudevent";
 import { ValidationError } from "./event/validation";
-import { CloudEventV03, CloudEventV03Attributes, CloudEventV1, CloudEventV1Attributes } from "./event/interfaces";
+export * from "./event/interfaces";
```
Comment (grant) [inline src/index.ts:3]: No star imports please.

File: examples/express-ex/index.js [hunk 2]
```diff
@@ -3,8 +3,8 @@
 const express = require("express");
 const { Receiver } = require("cloudevents");
 const app = express();
-const bodyParser = require('body-parser')
-app.use(bodyParser.json())
+const bodyParser = require("body-parser");
```
Comment (grant) [inline examples/express-ex/index.js:6]: I would not expect this change in this PR.

Was this fixed with some linting tool? Perhaps we should add it as a GitHub action step to check this folder.

Can we revert this file?

**Review by grant — APPROVED** [review]
A few requested changes. O.w. lgtm.

File: src/index.ts (hunk 1 above)
Comment (loopingz) [reply src/index.ts:3]: I think it makes more sense to export all interfaces directly, what is the reason behind no star in this case?

File: examples/express-ex/index.js (hunk 2 above)
Comment (loopingz) [reply examples/express-ex/index.js:6]: Yep

**Review by lance — COMMENTED** [review]
LGTM, though I'm still unsure whether these should live here or in a future `cloudevents-discovery` module created in this repo.

**Comment by loopingz** [comment]
It could be both but Discovery and Subscription might have a common definition, that is why I think SDK holding the definition might be a good way, as it is defining the common vocabulary (interfaces).
But at the same time, they might evolve more, it is a broader issue is cloudevents spec is the cloudevent/discovery/subscription or just cloudevent.

**Comment by lance** [comment]
Hmm yes. If the Subscription API and Discovery API each make use of types from the other, then I agree it makes sense for those types to live here.

File: src/index.ts (hunk 1 above)
Comment (grant) [reply src/index.ts:3]: Yeah, this looks fine actually!

**Review by grant — CHANGES_REQUESTED** [review]
It looks like this PR is out of date.

I suggest at this moment we encourage development of discovery features in a separate discovery-centric repo.

Removing approval since this repo has changed. Ping for re-review. :)

**Pull request closed without merging**